components/board/genericlinux/board.go

```diff
@@ -40,7 +41,7 @@ type Config struct {
 }
 
 // RegisterBoard registers a sysfs based board of the given model.
-func RegisterBoard(modelName string, gpioMappings map[int]GPIOBoardMapping) {
+func RegisterBoard(modelName string, gpioMappings map[int]GPIOBoardMapping, usePeriph bool) {
```

Comment:
what is the intention behind how `usePeriph` is meant to be used? 
If I understand the plan correctly, it'll only be used to control whether use it for gpio, but we continue to use it for serial, spi, etc.
0. we should document exactly what that boolean is meant to control
1. It feels fragile to rely on a number of properly placed if-statements to short circuit behavior in methods like GPIOPinByName...I think something like having GPIOPinbyName deletegate to GPIOPinByNamePeriph and GPIOPinByNameIoCtl would make that method less fragile and easier to parse.
2. If periph will be used for other purposes, this is a misleading name

Reply:
At the moment, we have 3 different implementations of GPIO: one using pigpio, one using periph, and a new one (indirectly) using ioctl. I'd prefer to eventually get down to a single implementation everywhere (or at least a single implementation in the genericlinux boards), which would be easier to think about, and have less code to maintain. So, I intend this variable to be a temporary thing lasting a few months, as we switch other boards over to using ioctl, too, and eventually they'll all be on it and we can remove this boolean.

I really like your point 2! What name should I use instead? I'd previously had it as `useIoctl` (with the values flipped), but that felt confusing because we don't use ioctl directly. The library that calls ioctl for us is named `gpio`, which is also not a good name to disambiguate this.

Reply:
Ack. I suspect it'll live for longer than a few months :). I ask that you pepper in some comments to describe the what and why. I'm fine with `useioctl` or `periphGpio`.